addons/account/tests/test_chart_template.py

```diff
@@ -38,160 +81,156 @@ def create_tax_template(cls, name, template_name, amount):
+        """
+        Tests that when we remove the tax from the fiscal position mapping it is not
+        recreated after update of taxes.
+        """
+        fiscal_position = self.env['account.fiscal.position'].search([('company_id', '=', self.company.id)])
         fiscal_position.tax_ids.unlink()
-        update_taxes_from_templates(self.env.cr, chart_template_xml_id)
-
-        # if only the fiscal position mapping has been removed, it won't be recreated
+        update_taxes_from_templates(self.env.cr, self.chart_template_xmlid)
         self.assertEqual(len(fiscal_position.tax_ids), 0)
 
     def test_update_taxes_conflict_name(self):
-        chart_template_xml_id = self.chart_template.get_external_id()[self.chart_template.id]
-        template_vals = self.tax_1_template._get_tax_vals_complete(self.company_1)
-        template_vals['amount'] = 20
-        self.chart_template.create_record_with_xmlid(self.company_1, self.tax_1_template, "account.tax", template_vals)
-        update_taxes_from_templates(self.env.cr, chart_template_xml_id)
-        tax_1_old = self.env['account.tax'].search([('company_id', '=', self.company_1.id), ('name', '=', "[old] " + self.tax_1_template.name)])
-        tax_1_new = self.env['account.tax'].search([('company_id', '=', self.company_1.id), ('name', '=', self.tax_1_template.name)])
+        """
+        When recreating a tax during update a conflict name can occur since
+        we need to respect unique constraint on (name, company_id, type_tax_use, tax_scope).
+        To do so, the old tax needs to be prefixed with '[old] '.
+        """
+        # We increment the amount so the template gets slightly different from the
+        # corresponding tax and triggers recreation
+        self.tax_template_1.amount += 1
+        update_taxes_from_templates(self.env.cr, self.chart_template_xmlid)
+        tax_1_old = self.env['account.tax'].search([('company_id', '=', self.company.id), ('name', '=', "[old] " + self.tax_template_1.name)])
+        tax_1_new = self.env['account.tax'].search([('company_id', '=', self.company.id), ('name', '=', self.tax_template_1.name)])
         self.assertEqual(len(tax_1_old), 1, "Old tax still exists but with a different name.")
         self.assertEqual(len(tax_1_new), 1, "New tax have been created with the original name.")
+
+    def test_update_taxes_multi_company(self):
+        """
+        In a multi-company environment all companies should be correctly updated.
+        """
+        company_2 = self.env['res.company'].create({
+            'name': 'TestCompany2',
+            'country_id': self.env.ref('base.be').id,
+            'account_tax_fiscal_country_id': self.env.ref('base.be').id,
+        })
+        self.chart_template.try_loading(company=company_2)
+
+        taxes_before = self.env['account.tax'].search([('name', '=', self.tax_template_1.name)])
+        self.assertEqual(taxes_before.mapped('amount'), [self.tax_template_1.amount] * len(taxes_before))
```

Comment:
This is not very clear. We should use an `assertRecordValues` instead and maybe add a comment about what you are testing here **if** the change is not clear enough.